src/nvim/tui/tui.c

```diff
@@ -845,7 +845,7 @@ CursorShape tui_cursor_decode_shape(const char *shape_str)
 
 static cursorentry_T decode_cursor_entry(Dictionary args)
 {
-  cursorentry_T r;
+  cursorentry_T r = { 0 };
```

Comment:
My suggestion here wasn't good, it fails quickbuild CI due to a (valid) warning. Maybe this instead:

    cursorentry_T r = shape_table[0];